app/models/mixins/container_resource_mixin.rb

```diff
@@ -0,0 +1,101 @@
+# ContainerResourceMixin provides capabilities to container
+# resources.
+module ContainerResourceMixin
+  extend ActiveSupport::Concern
+
+  def spec
+    resource = provider_definition
+    unless resource[:spec].nil?
+      return resource[:spec]
+    end
+    nil
+  end
+
+  def spec=(value)
+    resource = provider_definition
+    unless resource[:spec].nil?
+      resource[:spec] = value
+      response = update_in_provider(resource)
+      return response
+    end
+    nil
+  end
+
+  def annotations(annotation = nil)
+    resource = provider_definition
+    if annotation.nil?
+      return resource[:metadata][:annotations]
+    end
+    resource[:metadata][:annotations][annotation.to_sym]
+  end
+
+  def kind_in_provider
+    MIQ_CLASS_MAPPING[self.class.name]
```

Comment:
I've made an attempt at this in https://github.com/ManageIQ/manageiq-providers-kubernetes/pull/18.